### PR TITLE
Rewrote all queries as model.find_by_sql with bound params.

### PR DESCRIPTION
**Pull request by jasiek**

### CODE REVIEW

File: lib/detailed_metrics.rb [hunk 1]
```diff
@@ -18,9 +20,13 @@ def time_since_last_unprocessed
...
   def week_hour_breakdown(column)
+    column.define_singleton_method(:quoted_id) do
+      column
+    end
+
     @scoped_model.
-      group("EXTRACT(dow FROM #{column})::integer").
-      group("EXTRACT(hour FROM #{column})::integer").
+      group("EXTRACT(dow FROM #{quote(column)})::integer").
+      group("EXTRACT(hour FROM #{quote(column)})::integer").
```
Comment (SteveMarshall) [inline lib/detailed_metrics.rb]: Does group not accept parameterised input? Annoying. This looks good, though.

File: lib/calculated_metrics.rb [hunk 2]
```diff
@@ -44,32 +45,39 @@ def percent_rejected(prison, reason=nil)
     end
   end
 
-  def calculate_percentiles(column, percentile)
+  private
+  def calculate_percentiles(column, percentile) 
+    # This is a truly horrible hack. It lets us get past AR built-in value quoting, by pretending to be another object.
+    # It is needed so that we can parametrize two queries below by column, without resorting to putting the query together
```
Comment (SteveMarshall) [inline lib/calculated_metrics.rb]: Minor typo: `parametrize`.

File: lib/calculated_metrics.rb [hunk 3]
```diff
@@ -44,32 +45,39 @@ def percent_rejected(prison, reason=nil)
...
+    # by hand. Don't call this method from outside this class, please.
```
Comment (SteveMarshall) [inline lib/calculated_metrics.rb]: I dunno if we need the last sentence: does the `private` not imply that?

**Comment by SteveMarshall** [comment]
A few super-minor comments. Other than those, this looks great :+1:

File: lib/detailed_metrics.rb (hunk 1 above)
Comment (jasiek) [inline lib/detailed_metrics.rb]: Nope.

**Pull request merged**